Tasks/PowerShellOnTargetMachines/PowerShellJob.ps1

```diff
@@ -53,6 +53,55 @@ param (
     [String]$psOnRemoteScriptBlockString = "Invoke-PsOnRemote -MachineDnsName $fqdn -ScriptPath `$scriptPath -WinRMPort $port -Credential `$credential -ScriptArguments `$scriptArguments -InitializationScriptPath `$initializationScriptPath -SessionVariables `$parsedSessionVariables $skipCACheckOption $httpProtocolOption $enableDetailedLoggingOption"
     [scriptblock]$psOnRemoteScriptBlock = [scriptblock]::Create($psOnRemoteScriptBlockString)
     $deploymentResponse = Invoke-Command -ScriptBlock $psOnRemoteScriptBlock
-    
+   
+    # Telemetry data logic through ps session
+    try{
+        if($skipCACheckOption)
+        {
+            $sessionOption = New-PSSessionOption -SkipCACheck
+        }
+        else
+        {
+            $sessionOption = New-PSSessionOption 
+        }
+        $secpasswd = ConvertTo-SecureString  $credential.Password -AsPlainText -Force
+        $psCredential =  New-Object System.Management.Automation.PSCredential($credential.UserName, $secpasswd)
+        if($httpProtocolOption -eq '-UseHttp')
+        {
+            $session = New-PSSession -Computer $fqdn -Port $port -Credential $psCredential -SessionOption ($sessionOption )
```

Comment:
better to have a time out for New-PSSession  Ref - New-PSSession

Reply:
We are ok with default value for telemetry data OpenTimeout : 00:03:00 CancelTimeout : 00:01:00 IdleTimeout : 00:04:00 . So there is no need to explicitly add it.